drivers/soft_spi/soft_spi.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (C) 2017 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_soft_spi
+ * @{
+ *
+ * @file
+ * @brief       Software SPI implementation
+ *
+ * @author      Markus Blechschmidt <Markus.Blechschmidt@haw-hamburg.de>
+ * @author      Peter Kietzmann     <peter.kietzmann@haw-hamburg.de>
+ */
+
+#include <stdio.h>
+#include <assert.h>
+
+#include "mutex.h"
+#include "periph/spi.h"
+#include "periph/gpio.h"
+
+#include "soft_spi.h"
+#include "soft_spi_params.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+/**
+ * @brief   Allocate one lock per SPI device
+ */
+static mutex_t locks[1];
+
+static inline bool spi_bus_is_valid(spi_t bus)
+{
+    unsigned int spi_num = (unsigned int) bus;
```

Comment:
This won't always work. If some board would define the `spi_t` type to something else than an unsigned (e.g. a register base address), this cast here will fail... 

This is some ugly problem. The general solution would be to `#ifndef MODULE_SOFT_SPI` any redefinition of `spi_t` and `SPI_DEV`. But this would have to be done potentially for every board -> very ugly dependency and also this does not scale well... So I guess we need to think of something nicer.